tools/src/tester/StricterMergedOpenApiSpec.ts

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+import { type OpenAPIV3 } from 'openapi-types'
+import { Logger } from '../Logger'
+import { SpecificationContext } from '../linter/utils';
+import { SchemaVisitor } from '../linter/utils/SpecificationVisitor';
+import OpenApiMerger from '../merger/OpenApiMerger';
+
+// An augmented spec with additionalProperties: false.
+export default class StricterMergedOpenApiSpec {
+  logger: Logger
+  file_path: string
+  protected _spec: OpenAPIV3.Document | undefined
+
+  constructor (specPath: string, logger: Logger = new Logger()) {
+    this.logger = logger
+    this.file_path = specPath 
+  }
+
+  spec (): OpenAPIV3.Document {
+    if (this._spec) return this._spec
+    const spec = (new OpenApiMerger(this.file_path, this.logger)).merge()    
+    const ctx = new SpecificationContext(this.file_path)
+    this.inject_additional_properties(ctx, spec)
+    this._spec = spec
+    return this._spec
+  }
+
+  private inject_additional_properties(ctx: SpecificationContext, spec: OpenAPIV3.Document) {
+    const visitor = new SchemaVisitor((_ctx, schema) => {
+      if ('properties' in schema) {
+        // causes any undeclared field in the response to produce an error
+        (schema as any).additionalProperties = {
+          not: true,
+          errorMessage: "property is not defined in the spec"
+        }
+      }
+    });
```

Comment:
This should check if the additionalProperties is null first.

Reply:
I also wonder if making `addiltionalProperties: false` (when it's null) will have the same effect anyway without `ajv-errors`?

Reply:
By default, with or without `ajv-errors`, `addiionalProperties: false` produces `(data/version must NOT have additional properties)` without specifying which property is missing. The missing properties are available in the `params` value returned from `validate`, but we want a useful error message.

```
[
  {
    instancePath: '/version',
    schemaPath: '#/additionalProperties',
    keyword: 'additionalProperties',
    params: { additionalProperty: 'distribution' },
    message: 'must NOT have additional properties'
  }
]
```